esmvaltool/diag_scripts/mlr/__init__.py

```diff
@@ -0,0 +1,837 @@
+    fraction_weights = iris.util.broadcast_to_shape(fraction_weights,
+                                                    cube.shape,
+                                                    tuple(coord_dims))
+
+    return fraction_weights
+
+
+def get_new_path(cfg, old_path):
+    """Convert old path to new diagnostic path.
+
+    Parameters
+    ----------
+    cfg : dict
+        Recipe configuration.
+    old_path : str
+        Old path.
+
+    Returns
+    -------
+    str
+        New diagnostic path.
+
+    """
+    basename = os.path.splitext(os.path.basename(old_path))[0]
+    new_path = get_diagnostic_filename(basename, cfg)
+    return new_path
+
+
+def get_squared_error_cube(ref_cube, error_datasets):
+    """Get array of squared errors.
+
+    Parameters
+    ----------
+    ref_cube : iris.cube.Cube
+        Reference cube (determines mask, coordinates and attributes of output).
+    error_datasets : list of dict
+        List of metadata dictionaries where each dictionary represents a single
+        dataset.
+
+    Returns
+    -------
+    iris.cube.Cube
+        Cube containing squared errors.
+
+    Raises
+    ------
+    ValueError
+        Shape of a dataset does not match shape of reference cube.
+
+    """
+    squared_error_cube = ref_cube.copy()
+
+    # Fill cube with zeros
+    squared_error_cube.data = np.ma.array(
+        np.full(squared_error_cube.shape, 0.0),
+        mask=np.ma.getmaskarray(squared_error_cube.data),
+    )
+
+    # Adapt cube metadata
+    if 'error' in squared_error_cube.attributes.get('var_type', ''):
+        if not squared_error_cube.attributes.get('squared'):
+            squared_error_cube.var_name += '_squared'
+            squared_error_cube.long_name += ' (squared)'
+            squared_error_cube.units = units_power(squared_error_cube.units, 2)
+    else:
+        if squared_error_cube.attributes.get('squared'):
+            squared_error_cube.var_name += '_error'
+            squared_error_cube.long_name += ' (error)'
+        else:
+            squared_error_cube.var_name += '_squared_error'
+            squared_error_cube.long_name += ' (squared error)'
+            squared_error_cube.units = units_power(squared_error_cube.units, 2)
+    squared_error_cube.attributes['squared'] = 1
+    squared_error_cube.attributes['var_type'] = 'prediction_output_error'
+
+    # Aggregate errors
+    filenames = []
+    for dataset in error_datasets:
+        path = dataset['filename']
+        cube = iris.load_cube(path)
+        filenames.append(path)
+
+        # Check shape
+        if cube.shape != ref_cube.shape:
+            raise ValueError(
+                f"Expected shape {ref_cube.shape} for error cubes, got "
+                f"{cube.shape} for dataset '{path}'")
+
+        # Add squared error
+        new_data = cube.data
+        if not cube.attributes.get('squared'):
+            new_data **= 2
+        squared_error_cube.data += new_data
```

Comment:
I could, but since large parts of this code use `sklearn` (which can't handle `dask` at the moment) I don't think that this would change anything. The `sklearn` parts need all data realized anyway :smile:

Reply:
ah crap, ok then :+1: